apps/meteor/ee/server/lib/deviceManagement/session.ts

```diff
@@ -0,0 +1,121 @@
+import { Meteor } from 'meteor/meteor';
+import { Accounts } from 'meteor/accounts-base';
+import { UAParser } from 'ua-parser-js';
+import { ISocketConnection, IUser } from '@rocket.chat/core-typings';
+
+import * as Mailer from '../../../../app/mailer';
+import { Users } from '../../../../app/models/server/raw/index';
+import { settings } from '../../../../app/settings/server';
+import { UAParserDesktop, UAParserMobile } from '../../../../app/statistics/server/lib/UAParserCustom';
+import { sauEvents } from '../../../../server/services/sauMonitor/events';
+import { hasLicense } from '../../../app/license/server/license';
+import { t } from '../../../../app/utils/server';
+
+let mailTemplates: string;
+
+Meteor.startup(() => {
+	Mailer.getTemplate('Device_Management_Email_Body', (template) => {
+		mailTemplates = template;
+	});
+});
+
+const uaParser = async (
+	uaString: ISocketConnection['httpHeaders']['user-agent'],
+): Promise<UAParser.IResult & { app?: { name: string; version: string; bundle: string } }> => {
+	let rcAgent = {};
+	if (uaString && UAParserMobile.isMobileApp(uaString)) {
+		rcAgent = UAParserMobile.uaObject(uaString);
+	}
+
+	if (uaString && UAParserDesktop.isDesktopApp(uaString)) {
+		rcAgent = UAParserDesktop.uaObject(uaString);
+	}
+
+	const ua = new UAParser(uaString);
+	return { ...ua.getResult(), ...rcAgent };
+};
+
+export const listenSessionLogin = async (): Promise<void> => {
+	sauEvents.on('accounts.login', async ({ userId, connection }) => {
+		if (!hasLicense('device-management')) return;
+
+		const user = await Users.findOneById<IUser>(userId, { projection: { name: 1, username: 1, emails: 1 } });
+		if (user?.emails?.length && !connection.loginToken) {
+			const {
+				name,
+				username,
+				emails: [{ address: email }],
+			} = user;
+			const { browser, os, device, cpu, app } = await uaParser(connection.httpHeaders['user-agent']);
+
+			let mailData = {};
```

Comment:
I saw a lot of duplicated code when defining a new value for `mailData` in the `switch` below, instead I'd recommend to change to a `const` and provide the common values here and just provide the new value on the switch statement, something like this:
```suggestion
			const mailData = {
				name,
				username,
				ipInfo: connection.clientAddress,
			};
```